msal/tests/Test.MSAL.NET.Integration/DeviceCodeIntegrationTests.cs

```diff
@@ -47,6 +47,7 @@ public class DeviceCodeIntegrationTests
 
         [TestMethod]
         [TestCategory("UsernamePasswordIntegrationTests")]
+        [Ignore]
```

Comment:
I don't think this is needed, they should pass on the build machine as the user running the tests has access to KeyVault. @jennyf19 - the tests should fail more gracefully, i.e. they fail with a NullRef today, they should fail with a nice message

Reply:
this test more some kind of manual test since the step of authentication is not automated, so it is not integration test

Reply:
pls. do not ignore tests. if it doesn't pass locally due to missing setup investigate how to best fix that.

---
In reply to: [222927993](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1304#discussion_r222927993) [](ancestors = 222927993)

Reply:
it does not work because authentication is not implemented, so it is not automated test

Reply:
UsernamePassword is implemented and shipped on MSAL since 2.1.0. We even shipped Device Flow in 2.2.0 :). This test passes on the build machine, but fails on our machines. If you just delete the [Ignore] I am sure it will not cause issues on the build machine. We will look separately into making it pass or at least fail more gracefully on dev machines.

Reply:
it can not pass anywhere it is not even test , it makes no sense, it just some random copy-paste(test category, test name is not related to  device code flow, user object is not used , empty call back is passed)
It just wait 15 mins and than fails (because deviceCode result is just ignored - empty callback passed).

If you think it pass on build machine It just means that it is not get run there.

Reply:
If I understood correctly, this is a copy paste error. Please delete the test. 

In general we should not [Ignore] tests because we forget about them and after a few months they become dead code. We should only Ignore if we create a bug / work item to track the work of un-ignoring.

Reply:
@mzuber@microsoft.com : What was the intentions of this? Should it be deleted? How is the e2e of Device Codeflow validated now? Manually (does the test exist in VSTS)?

---
In reply to: [224132337](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1304#discussion_r224132337) [](ancestors = 224132337)

Reply:
This was put in early in development of devicecode and wasn't fully fleshed out because we can't automate the browser side of things.  so i'll kill this test for now.  yes, Henrik, for now this test is manual since we don't automate the browser.

leave it as ignore in your PR (it's ignore on my end too) and i'll kill it with my refactor PR.